Source/WebCore/bindings/js/JSDOMConvertStrings.cpp

```diff
@@ -118,4 +121,94 @@ AtomString valueToUSVAtomString(JSGlobalObject& lexicalGlobalObject, JSValue val
     return replaceUnpairedSurrogatesWithReplacementCharacter(WTFMove(string));
 }
 
+String valueToTrustedHTMLString(JSGlobalObject& lexicalGlobalObject, JSValue value, const String& sink, bool nullToEmpty)
+{
+    UNUSED_PARAM(sink);
+    VM& vm = lexicalGlobalObject.vm();
+
+    if (auto* trustedHTML = JSTrustedHTML::toWrapped(vm, value))
+        return trustedHTML->toString();
+
+    // TODO: Implement enforcement logic
```

Comment:
Also, we use `FIXME`, not `TODO`.

Reply:
Is the enforcement particularly complicated? I was wondering why we don’t have a first cut at the enforcement logic in this initial patch.

Reply:
Yeah the enforcement involves checking CSP and calling into default policies in specific cases. I could do it in one go but previously was told to keep IDL changes separate to keep PRs smaller.

Reply:
This seems to do a bit too little; like at least we should have the enforcement function even if it has a placeholder simplistic rule, instead of many different `TODO` comments.

Reply:
I've added some of the structure of the enforcement but it will require CSP changes to implement fully (have a follow up for that)

Reply:
@darinadler the CSP parsing is merged now so I could go further and actually include the CSP enforcement code too if it's preferable for this to be done in one hit? Unless I also apply the [StringContext] IDL attribute to some of the sinks in this PR the CSP enforcement won't have test coverage though?